### ISSUE TITLE
Command passed raises Unkown typ

### ISSUE THREAD

**Issue by diamonedge** [paramiko/paramiko#937]
I came around this issue : 
```bash
  File "/usr/local/lib/python2.7/site-packages/paramiko/client.py", line 441, in exec_command
    chan.exec_command(command)
  File "/usr/local/lib/python2.7/site-packages/paramiko/channel.py", line 60, in _check
    return func(self, *args, **kwds)
  File "/usr/local/lib/python2.7/site-packages/paramiko/channel.py", line 231, in exec_command
    m.add_string(command)
  File "/usr/local/lib/python2.7/site-packages/paramiko/message.py", line 285, in add_string
    s = asbytes(s)
  File "/usr/local/lib/python2.7/site-packages/paramiko/common.py", line 170, in asbytes
    raise Exception('Unknown type')
Exception: Unknown type
```
if it helps somebody, you might convert passed command and username/password using : 
```python
self.__ssh = paramiko.SSHClient()
self.__ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
self.__ssh.connect(self.__hostname, username=str(self.__username), password=str(self.__password), port=self.__port)
stdin, stdout, stderr = self.__ssh.exec_command(str(command))
```
Converting passed var to ```str()``` did the trick on my side, yet is this legit ?

**Comment by ploxiln** [comment]
What type was command before before converting it with `str()`? Can you give any more information about what it was, where it came from, how it got there?

**Comment by bz2** [comment]
This is fixed as side-effect of #971 by changing the `asbytes` implementation. Users should get a more useful `TypeError` from further down the stack instead.
